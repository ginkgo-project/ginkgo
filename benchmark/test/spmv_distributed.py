#!/usr/bin/env python3
import test_framework

# check that all input modes work:
# parameter
test_framework.compare_output_distributed(
    ["-input", '[{"size": 100, "stencil": "7pt", "comm_pattern": "stencil"}]'],
    expected_stdout="spmv_distributed.simple.stdout",
    expected_stderr="spmv_distributed.simple.stderr",
    num_procs=3,
)

# stdin
test_framework.compare_output_distributed(
    [],
    expected_stdout="spmv_distributed.simple.stdout",
    expected_stderr="spmv_distributed.simple.stderr",
    num_procs=3,
    stdin='[{"size": 100, "stencil": "7pt", "comm_pattern": "stencil"}]',
)

# input file
test_framework.compare_output_distributed(
    ["-input", str(test_framework.sourcepath / "input.distributed_mtx.json")],
    expected_stdout="spmv_distributed.simple.stdout",
    expected_stderr="spmv_distributed.simple.stderr",
    num_procs=3,
)

# profiler annotations
test_framework.compare_output_distributed(
    [
        "-input",
        '[{"size": 100, "stencil": "7pt", "comm_pattern": "stencil"}]',
        "-profile",
        "-profiler_hook",
        "debug",
    ],
    expected_stdout="spmv_distributed.profile.stdout",
    expected_stderr="spmv_distributed.profile.stderr",
    num_procs=3,
)

# complex
test_framework.compare_output_distributed(
    ["-input", '[{"size": 100, "stencil": "7pt", "comm_pattern": "stencil"}]'],
    expected_stdout="spmv_distributed_dcomplex.simple.stdout",
    expected_stderr="spmv_distributed_dcomplex.simple.stderr",
    num_procs=3,
    use_complex=True
)
