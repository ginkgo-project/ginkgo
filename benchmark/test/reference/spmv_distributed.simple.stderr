Running on reference(0)
Running with 2 warm iterations and 10 running iterations
The random seed for right hand sides is 42
The formats are [csr]x[csr]
The number of right hand sides is 1
Running test case stencil(100, 7pt, stencil)
Matrix is of size (81, 81), 144
	Running spmv: csr-csr
